### PR TITLE
Ajc add cell barcode correction

### PR DESCRIPTION
**Pull request by ambrosejcarr**
This PR contains a full Optimus MVP (implemented according to the [Design Specification](https://docs.google.com/document/d/12TutsWTYkc9wAJar-16lJveQ4CXMh5nwL0C8DaG3A4E/edit?usp=sharing), confirmed to run on test data. 

It adds:

WDL:
- Cell barcode correction & splitting
- A sub-workflow enabling a nested scatter over:
  - Alignment, Gene Tagging, UMI & Duplicate Correction
- Array[Array[bam]] -> bam Merge task
- Picard CollectMultipleMetrics
- Count Matrix Construction (coordinate sparse)

Dockers:
- Dropseqtools
- Samtools

Testing notebooks:
- test_optimus.ipynb

Where possible, existing software (picard, dropseq tools) are used. 

There is a related PR in sctools for [Bam Splitting](https://github.com/HumanCellAtlas/sctools/pull/4)

### CODE REVIEW

File: docker/samtools/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+FROM ubuntu:zesty
+
+LABEL maintainer="Ambrose J. Carr <acarr@broadinstitute.org>" \
+  software="samtools" \
+  version="1.3.1" \
+  description="processing sequence alignments in SAM and BAM formats" \
+  website="https://samtools.github.io"
+
+RUN apt update && apt install -y \
+  samtools
```
Comment (rexwangcc) [inline docker/samtools/Dockerfile]: Sorry I'm a little bit confused here, how do you make sure you are installing the version 1.3.1 of samtools? I saw there the latest version is 1.6 now: (https://github.com/samtools/samtools/tags).

File: pipelines/tasks/CollectMultiplePicardMetrics.wdl [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      OUTPUT="${output_filename}" \
+      PROGRAM=CollectAlignmentSummaryMetrics \
+      PROGRAM=CollectInsertSizeMetrics \
+      PROGRAM=CollectGcBiasMetrics \
+      REFERENCE_SEQUENCE="${ref_genome_fasta}"
+
+      tar -czvf "${output_filename}.tar.gz" "${output_filename}"*
+  }
+  runtime {
+    docker:"humancellatlas/picard:2.10.10"
```
Comment (rexwangcc) [inline pipelines/tasks/CollectMultiplePicardMetrics.wdl]: Don't know if you want the R-dependencies, if you need, `humancellatlas/picard:2.10.10-7ab16db` is the one with R-base and ggplot2 installed.

File: docker/samtools/Dockerfile (hunk 1 above)
Comment (ambrosejcarr) [reply docker/samtools/Dockerfile]: Good catch; I'm not sure, actually -- but 1.3.1 is the version that gets installed by apt. I will specify the version explicitly.

File: smartseq2_single_sample/tasks/picard.wdl [hunk 3]
```diff
@@ -17,7 +17,7 @@ task CollectMultipleMetrics {
       REFERENCE_SEQUENCE=${ref_genome_fasta} \
       ASSUME_SORTED=true
 
-      tar -cvf "${output_file}.tar" ${output_file}.*.txt
+      tar -cvf "${output_filename}.tar" ${output_filename}.*.txt
```
Comment (jishuxu) [inline smartseq2_single_sample/tasks/picard.wdl]: the PR#50 will rewrite this section. basically, I am keeping individual files as outputs without tar them.

**Comment by jishuxu** [comment]
it is fine to merge.

File: pipelines/tasks/CollectMultiplePicardMetrics.wdl (hunk 2 above)
Comment (ambrosejcarr) [reply pipelines/tasks/CollectMultiplePicardMetrics.wdl]: I am going to pull Jishu's task in after she merges her picard task; at that point I will change the docker.

File: smartseq2_single_sample/tasks/picard.wdl (hunk 3 above)
Comment (ambrosejcarr) [reply smartseq2_single_sample/tasks/picard.wdl]: Great. I look forward to changing it!

**Pull request merged**